firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteStore.java

```diff
@@ -369,7 +369,14 @@ public void listen(TargetData targetData) {
 
   private void sendWatchRequest(TargetData targetData) {
     watchChangeAggregator.recordPendingTargetRequest(targetData.getTargetId());
-    watchStream.watchQuery(targetData);
+    if (!targetData.getResumeToken().isEmpty()
+        || targetData.getSnapshotVersion().compareTo(SnapshotVersion.NONE) > 0) {
+      int expectedCount = this.getRemoteKeysForTarget(targetData.getTargetId()).size();
+      TargetData newTargetData = targetData.withExpectedCount(expectedCount);
```

Comment:
Please make this consistent with the web sdk as well. Namely, replace the `targetData` local variable rather than creating the `newTargetData` local variable. Also, move the call to `watchStream.watchQuery(targetData)` to _outside_ of the "if" block.

https://github.com/firebase/firebase-js-sdk/blob/dd6683583b7c6b07586c2551f4d362b48fb83f7c/packages/firestore/src/remote/remote_store.ts#L338-L346